### DISCUSSION TITLE
android targetSdkVersion, 30 or 31?

### DISCUSSION THREAD

**Discussion by jimmymjing** [airsdk/Adobe-Runtime-Support#1842] [Q&A]
Hello, as most of the ANEs from Distriqt have been updated, shall we change the targetSdkVersion to 31 or keep it as 30? Thanks!

**Comment by shaymolho** [comment]
@marchbold @ajwfrost please advise

**Reply by ajwfrost** [reply]
Hi
See https://developer.android.com/google/play/requirements/target-sdk
 > Starting in August 2022, new apps must target API level 31 (Android 12) or above and adjust for behavioral changes.
> Starting in November 2022, app updates must target API level 31 or above and adjust for behavioral changes in Android 12.

thanks

**Reply by marchbold** [reply]
Yeah definitely start targeting 31 now. We've run a heap of tests on our extensions on 31, but let us know if you run into anything.

**Reply by shaymolho** [reply]
Is there anything else that needs to be done other than changing the `targetSdkVersion` xml element to 31?
Do I need to re-run apm to generate the app-descriptor?
I read the migration guide but I was wondering if apm already handles it

**Reply by marchbold** [reply]
`apm` doesn't automatically insert the target sdk version (unless some of the libs in extensions you are using set it), however if you do want it to control it, you can generate a custom config file:

```
apm generate config android
```

Then edit the generated `config/android/AndroidManifest.xml` file. The default should have the tag for the target sdk and you will likely want to change it. Once you've done this, `apm` will merge the contents of that file with the rest of the additions from extensions etc. 

```
apm generate app-descriptor MyApp-app.xml
```
